gammapy/spectrum/results.py

```diff
@@ -355,161 +392,175 @@ def to_sherpa_model(self, name='default'):
 
         return model
 
-    def plot(self, ax=None, energy_unit='TeV',
-             flux_unit='cm-2 s-1 TeV-1', e_power=0, **kwargs):
-        """Plot fit function
+    def evaluate(self, x):
+        """Wrapper around astropy.modelling.models.evaluate
 
-        kwargs are forwarded to :func:`~matplotlib.pyplot.errorbar`
+        Parameters
+        ----------
+        x : `~gammapy.utils.energy.Energy`
+            Evaluation point
+        """
+        if self.spectral_model == 'PowerLaw':
+            flux = models.PowerLaw1D.evaluate(x, self.parameters.norm,
+                                              self.parameters.reference,
+                                              self.parameters.index)
+        elif self.spectral_model == 'LogParabola':
+            #LogParabola evaluation does not work with arrays
```

Comment:
What!?

Can you post a minimal code snippet here that shows that it doesn't work?

Leave a TODO here or directly file an issue or PR with Astropy?